File: JavaSelenium/src/test/java/com/taskmanager/e2e/pages/HeaderPage.java
package com.taskmanager.e2e.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class HeaderPage extends BasePage {

  private static final String LOGIN_BUTTON_TEST_ID = "open-login-button";

  public HeaderPage(WebDriver driver) {
    super(driver);
  }

  private WebElement loginButton() {
    return driver.findElement(Locators.testId(LOGIN_BUTTON_TEST_ID));
  }

  public void openLogin() {
    loginButton().click();
  }
}

File: APIandUItesting/apps/web-e2e/pages/header_page.ts
import { expect } from '../tests/test_base';
import { BasePage } from './base_page';

const HEADER_PAGE_TEST_IDS = {
  loginButton: 'open-login-button',
  logoutButton: 'logout-button',
};

export class HeaderPage extends BasePage {
  loginButton = this.page.getByTestId(HEADER_PAGE_TEST_IDS.loginButton);
  logoutButton = this.page.getByTestId(HEADER_PAGE_TEST_IDS.logoutButton);

  openLogin(): Promise<void> {
    return this.loginButton.click();
  }

  async logout(): Promise<void> {
    await expect(this.logoutButton).toBeVisible();

    await this.logoutButton.click();
  }
}
